crates/wash-lib/src/registry.rs

```diff
@@ -255,11 +264,14 @@ pub async fn push_oci_artifact(
 
 /// Helper function to determine artifact type and parse it into a config and layer ready for use in
 /// pushing to OCI
-pub async fn parse_and_validate_artifact(artifact: Vec<u8>) -> Result<SupportedArtifacts> {
+pub async fn parse_and_validate_artifact(
+    artifact: Vec<u8>,
```

Comment:
Do you need to take ownership of the vector here?

Reply:
Well, I did need it before but then I needed to clone anyway. I'll change that